internal/discovery/client.go

```diff
@@ -26,20 +26,20 @@ type DiscoveryClientConfig struct {
 	PollClusterInfoInterval time.Duration
 }
 
-func NewDiscoveryClientConfigFromIni(sect *ini.Section) (*DiscoveryClientConfig, error) {
-	sectConf := sect.KeysHash()
-	if discoveryServiceAddr, ok := sectConf["discoveryServiceAddr"]; ok {
-		if pushStatusInterval, err := strconv.Atoi(sectConf["pushStatusInterval"]); err == nil {
-			if pollClusterInfoInterval, err := strconv.Atoi(sectConf["pollClusterInfoInterval"]); err == nil {
+func ValidateAndGetDiscoveryClientConfig(discoveryClientConfigDto dtos.DiscoveryClientConfigDto) (*DiscoveryClientConfig, error) {
+
+	if len(discoveryClientConfigDto.DiscoveryServiceAddr) > 0 {
+		if pushStatusInterval, err := strconv.Atoi(discoveryClientConfigDto.PushStatusInterval); err == nil {
+			if pollClusterInfoInterval, err := strconv.Atoi(discoveryClientConfigDto.PollClusterInfoInterval); err == nil {
 				return &DiscoveryClientConfig{
-					DiscoveryServiceAddr:    discoveryServiceAddr,
+					DiscoveryServiceAddr:    discoveryClientConfigDto.DiscoveryServiceAddr,
 					PushStatusInterval:      time.Duration(pushStatusInterval),
 					PollClusterInfoInterval: time.Duration(pollClusterInfoInterval),
 				}, nil
 			}
 		}
 	}
-	return nil, fmt.Errorf("Invalid discovery client configuration. Check section %s", sect.Name())
+	return nil, fmt.Errorf("Invalid discovery client configuration")
```

Comment:
This seems redundant.  This can be completely avoided by defining proper data type for each of the params.

Reply:
This had been added because if we use datatype directly, mapstructure is not handling type validations and converts a negative number to overflowed uint number.